### PR TITLE
Monorepo structure for allowing usage of components outside of this project

### PR DESCRIPTION
**Pull request by lauriii**
## Issue

https://github.com/jsdrupal/drupal-admin-ui/issues/497

## Description
This modifies the directory structure so that the project is set up as monorepo. I used yarn workspaces but we could consider using [lerna](https://github.com/lerna/lerna) if we find their tooling useful. The packages are under `packages/` folder. The new `drupal-ui` package will contain all of the reusable components.

To make sure that at least the build works, I created an entry point which contains all of the components inside the component library. We might want to modify this so that components can be loaded one by one.

## Remaining tasks
- Update documented installation steps to match with the new structure
- Update installation steps used in the CI to match with the documented steps
- Allow loading of individual components.

## Example of installation steps for developers wanting to use the component library
*Note: these steps won't work before the package is published in the npm registry*

Add dependency for the package 🚀
`yarn add drupal-ui`

Use components 👯‍♀️🎉
```
import { PageTitle } from 'drupal-ui';

const app = () => (
  <div>
    <PageTitle>Title</PageTitle>
  </div>
)
```

### CODE REVIEW

File: packages/drupal-admin-ui/src/components/05_pages/AddContent/AddContent.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ import Paper from '@material-ui/core/Paper';
 import List from '@material-ui/core/List';
 import ListItem from '@material-ui/core/ListItem';
 import ListItemText from '@material-ui/core/ListItemText';
-import PageTitle from '../../02_atoms/PageTitle';
+import { PageTitle } from 'drupal-ui';
```
Comment (dawehner) [inline packages/drupal-admin-ui/src/components/05_pages/AddContent/AddContent.js]: I love how this makes things actually more readable!

File: packages/drupal-ui/build/InlineMessage/InlineMessage.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+var _modifierClassMap;
```
Comment (dawehner) [inline packages/drupal-ui/build/InlineMessage/InlineMessage.js]: We  should not commit the build folder

File: packages/drupal-admin-ui/src/components/02_atoms/Widgets/BooleanCheckbox.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ import PropTypes from 'prop-types';
 import CheckBox from '@material-ui/core/Checkbox';
 import FormControlLabel from '@material-ui/core/FormControlLabel';
 import { css } from 'emotion';
-import WidgetPropTypes from '../../../drupal-admin-ui/src/components/05_pages/NodeForm/WidgetPropTypes';
```
Comment (dawehner) [inline packages/drupal-admin-ui/src/components/02_atoms/Widgets/BooleanCheckbox.js]: I'm wondering whether it's a valid usecase to try to reuse these widgets?

Comment (lauriii) [reply packages/drupal-admin-ui/src/components/02_atoms/Widgets/BooleanCheckbox.js]: It seems like these widgets are more about functionality than the visual looks than the rest of the components so we probably should look at them separately at some point.

**Comment by lauriii** [comment]
It seems like the `lint_unit` test is failing because of there's something wrong with CircleCI itself. I run the task using CircleCI locally and the task is passing. My local installation didn't support caches, so I would guess this problem to be most likely caused by that.

**Comment by justafish** [comment]
Moved to #515

**Pull request closed without merging**